utils/src/main/java/software/amazon/awssdk/utils/CompletableFutureUtils.java

```diff
@@ -77,4 +77,53 @@ public static CompletionException errorAsCompletionException(Throwable t) {
         });
         return src;
     }
+
+    /**
+     * Returns a new CompletableFuture that is completed when any of
+     * the given CompletableFutures completes exceptionally.
+     *
+     * @param futures the CompletableFutures
+     * @return a new CompletableFuture that is completed if any provided
+     * future completed exceptionally.
+     */
+    public static CompletableFuture<Void> anyFail(CompletableFuture<?>[] futures) {
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+
+        for (CompletableFuture<?> future : futures) {
+            future.whenComplete((r, t) -> {
+                if (t != null) {
+                    completableFuture.complete(null);
+                }
+            });
+        }
+
+       return completableFuture;
+    }
+
+
+    /**
+     * Similar to {@link CompletableFuture#allOf(CompletableFuture[])}, but
+     * when any future is completed exceptionally, forwards the
+     * cancel to other futures.
+     *
+     * @param futures The futures.
+     * @return The new future that is completed when all the futures in {@code
+     * futures} are.
+     */
+    public static CompletableFuture<Void> allOfCancelForwarded(CompletableFuture<?>[] futures) {
+        CompletableFuture<?>[] futuresCopy = new CompletableFuture[futures.length];
+        System.arraycopy(futures, 0, futuresCopy, 0, futures.length);
+
+        CompletableFuture<Void> anyFail = anyFail(futuresCopy);
+
+        anyFail.whenComplete((r, t) -> {
+            for (CompletableFuture<?> cf : futuresCopy) {
+                if (!cf.isDone()) {
```

Comment:
This check isn't needed, since there's no guarantee that it will still be true when `cancel()` is called.